spotbugs/src/main/java/edu/umd/cs/findbugs/ba/ZipSourceFileDataSource.java

```diff
@@ -33,11 +34,14 @@
 
     private final String entryName;
 
+    private final URI entryURI;
+
     private final ZipEntry zipEntry;
 
     public ZipSourceFileDataSource(ZipFile zipFile, String entryName) {
         this.zipFile = zipFile;
         this.entryName = entryName;
+        this.entryURI = URI.create(entryName);
```

Comment:
I'm not sure this really works or not.
Both ZipFile and [ZipEntry](https://docs.oracle.com/javase/8/docs/api/java/util/zip/ZipEntry.html) provide no method returning `URI`, so I think that we need to use `FileSystems` like https://stackoverflow.com/q/37936627

Reply:
Hi @olegshtch, could you confirm this? Thank you!

Reply:
I've implemented it.